src/spyglass/decoding/v0/dj_decoder_conversion.py

```diff
@@ -0,0 +1,201 @@
+        ObservationModel,
+    )
+    from track_linearization import make_track_graph
+
+    def _convert_dict_to_class(d: dict, class_conversion: dict) -> object:
+        """Converts a dictionary into a class object
+
+        Parameters
+        ----------
+        d : dict
+        class_conversion : dict
+
+        Returns
+        -------
+        class_based_on_dict : object
+
+        """
+        class_name = d.pop("class_name")
+        return class_conversion[class_name](**d)
+
+    def _convert_env_dict(env_params: dict) -> Environment:
+        """If the track graph is in the environment parameters, convert it to a networkx graph
+
+        Parameters
+        ----------
+        env_params : dict
+
+        Returns
+        -------
+        environment : Environment
+        """
+        if env_params["track_graph"] is not None:
+            env_params["track_graph"] = make_track_graph(
+                **env_params["track_graph"]
+            )
+
+        return Environment(**env_params)
+
+    def _to_dict(transition: object) -> dict:
+        """Helper function to convert a transition class into a dictionary"""
+        parameters = vars(transition)
+        parameters["class_name"] = type(transition).__name__
+
+        return parameters
+
+    def _convert_transitions_to_dict(
+        transitions: list[list[object]],
+    ) -> list[list[dict]]:
+        """Converts a list of lists of transition classes into a list of lists of dictionaries"""
+        return [
+            [_to_dict(transition) for transition in transition_rows]
+            for transition_rows in transitions
+        ]
+
+    def restore_classes(params: dict) -> dict:
+        """Converts a dictionary of parameters into a dictionary of classes since datajoint cannot handle classes"""
+        continuous_state_transition_types = {
+            "RandomWalk": RandomWalk,
+            "RandomWalkDirection1": RandomWalkDirection1,
+            "RandomWalkDirection2": RandomWalkDirection2,
+            "Uniform": Uniform,
+            "Identity": Identity,
+        }
+
+        discrete_state_transition_types = {
+            "DiagonalDiscrete": DiagonalDiscrete,
+            "UniformDiscrete": UniformDiscrete,
+            "RandomDiscrete": RandomDiscrete,
+            "UserDefinedDiscrete": UserDefinedDiscrete,
+        }
+
+        initial_conditions_types = {
+            "UniformInitialConditions": UniformInitialConditions,
+            "UniformOneEnvironmentInitialConditions": UniformOneEnvironmentInitialConditions,
+        }
+
+        params["classifier_params"]["continuous_transition_types"] = [
+            [
+                _convert_dict_to_class(st, continuous_state_transition_types)
+                for st in sts
+            ]
+            for sts in params["classifier_params"][
+                "continuous_transition_types"
+            ]
+        ]
+        params["classifier_params"]["environments"] = [
```

Comment:
This looks like it could be a for loop
```
for attr, params in zip(['params', 'discrete_...'],[env_params, discrete_..]):
    params['classifier_params'][attr] = _convert(params['classifier_params'][attr], params)